src/snowflake/cli/api/commands/overridable_parameter.py

```diff
@@ -0,0 +1,143 @@
+        pass
+
+    class InvalidCallbackSignature(ClickException):
+        def __init__(self, callback):
+            super().__init__(
+                f"Signature {signature(callback)} is not valid for an OverrideableOption callback function. Must have "
+                f"at most one parameter with each of the following types: (typer.Context, typer.CallbackParam, "
+                f"Any Other Type)"
+            )
+
+    def _callback_factory(
+        self, callback, mutually_exclusive: Optional[List[str] | Tuple[str]]
+    ):
+        callback = callback if callback else lambda x: x
+
+        # inspect existing_callback to make sure signature is valid
+        existing_params = signature(callback).parameters
+        # at most one parameter with each type in [typer.Context, typer.CallbackParam, any other type]
+        limits = [
+            lambda x: x == typer.Context,
+            lambda x: x == typer.CallbackParam,
+            lambda x: x != typer.Context and x != typer.CallbackParam,
+        ]
+        for limit in limits:
+            if len([v for v in existing_params.values() if limit(v.annotation)]) > 1:
+                raise self.InvalidCallbackSignature(callback)
+
+        def generated_callback(ctx: typer.Context, param: typer.CallbackParam, value):
+            if mutually_exclusive:
+                for name in mutually_exclusive:
+                    if value and ctx.params.get(
+                        name, False
+                    ):  # if the current parameter is set to True and a previous parameter is also Truthy
+                        curr_opt = param.opts[0]
+                        other_opt = [x for x in ctx.command.params if x.name == name][
+                            0
+                        ].opts[0]
+                        raise IncompatibleParametersError([curr_opt, other_opt])
+
+            # pass args to existing callback based on its signature (this is how Typer infers callback args)
+            passed_params = {}
+            for existing_param in existing_params:
+                annotation = existing_params[existing_param].annotation
+                if annotation == typer.Context:
+                    passed_params[existing_param] = ctx
+                elif annotation == typer.CallbackParam:
+                    passed_params[existing_param] = param
+                else:
+                    passed_params[existing_param] = value
+            return callback(**passed_params)
+
+        return generated_callback
+
+
+class OverrideableArgument(_OverrideableParameter):
+    def get_parameter(
+        self, default: Any = ..., *param_decls: str, **kwargs
+    ) -> typer.models.ArgumentInfo:
+        return typer.Argument(default, *param_decls, **kwargs)
+
+
+# OverrideableOption doesn't work with flags with type List[Any] and default None. Because typer executes the callback
+# function and convert the default value iterating over it, but None is not iterable.
```

Comment:
```suggestion
# OverrideableOption doesn't work with flags with type List[Any] and default None, because typer executes the callback
# function which converts the default value iterating over it, but None is not iterable.
```